### PR TITLE
Scan: Revamp every state of confirmation popup

### LINKED ISSUE #2738: Revamp confimation modal

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #2738 

## Implementation details

- Implemented the confirmation modal
  - JSON preview
  - Success/Fail status
  - Loading state

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 1]
```diff
@@ -111,140 +53,149 @@ let make = (~rawTx, ~onBack, ~account: AccountContext.t) => {
 
   <div className=Styles.container>
     <div className=Styles.modalTitle>
-      <Text value="Confirm Transaction" weight=Text.Bold size=Text.Xxxl />
-    </div>
-    <VSpacing size=Spacing.lg />
-    <div className=Styles.warning>
-      <Text value="Please verify the transaction details below before proceeding" />
+      <Text value="Connect with your wallet" weight=Text.Medium size=Text.Xl />
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re]: Is it correct?

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 2]
```diff
@@ -111,140 +53,149 @@ let make = (~rawTx, ~onBack, ~account: AccountContext.t) => {
...
-           <Link className=Styles.txhashContainer route={Route.TxIndexPage(txHash)}>
+         <Link className=Styles.txhashContainer route={Route.TxIndexPage(txHash)}>
+           <Button py=8 px=13 variant=Button.Outline onClick={_ => {dispatchModal(CloseModal)}}>
              <Text
                block=true
-               code=true
-               spacing={Text.Em(0.02)}
-               value={txHash |> Hash.toHex(~upper=true)}
-               weight=Text.Medium
+               value="View Detail"
```
Comment (evilpeach) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re]: View Details?

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/cypress/integration/send_action.spec.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ describe("Send", () => {
       .get('[id="sendAmountInput"]')
       .type("2");
     cy.get('[id="nextButton"]').contains("Next").click();
-    cy.get('[id="broadcastButton"]').wait(1000).click();
+    cy.get('[id="broadcastButton"] > button').wait(1000).click();
```
Comment (Pitchanai) [inline scan/cypress/integration/send_action.spec.js]: Could you change the `id` to `broadcastButtonContainer`?

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 4]
```diff
@@ -111,140 +53,149 @@ let make = (~rawTx, ~onBack, ~account: AccountContext.t) => {
...
-                               };
-                             dispatchModal(EnableExit);
-                             Js.Promise.resolve();
-                           | _ =>
-                             setState(_ => Error("Fail to broadcast"));
+         <textarea
+           className=Styles.jsonDisplay
+           disabled=true
+           defaultValue={rawTx |> TxCreator.stringifyWithSpaces}
+         />
```
Comment (Pitchanai) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re:83]: Could you add a `font-family` to `<textarea />` to match the design?

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 5]
```diff
@@ -2,98 +2,40 @@ module Styles = {
...
-    style([
-      display(`flex),
-      flexDirection(`column),
-      alignItems(`center),
-      justifyContent(`center),
-      paddingTop(`px(15)),
-      minHeight(`px(70)),
-      maxHeight(`px(70)),
-    ]);
+  let modalTitle = style([paddingBottom(`px(16))]);
```
Comment (Pitchanai) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re]: Please change the `padding-bottom` to `24px`.

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 6]
```diff
@@ -111,140 +53,149 @@ let make = (~rawTx, ~onBack, ~account: AccountContext.t) => {
...
-           <HSpacing size=Spacing.md />
-           <Text value="Broadcast Transaction Success" weight=Text.Semibold />
+       <div
+         className={Css.merge([
+           CssHelper.flexBox(~direction=`column, ~justify=`center, ()),
+           Styles.resultContainer,
+         ])}>
+         <img src=Images.success className=Styles.resultIcon />
+         <div id="successMsgContainer" className={CssHelper.mb(~size=16, ())}>
+           <Text value="Broadcast Transaction Success" size=Text.Lg block=true align=Text.Center />
```
Comment (Pitchanai) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re]: Please check that the message not match with the design and don't have to be all caps.

File: scan/src/components/modal/submitTx/PreviewJsonStep.re [hunk 7]
```diff
@@ -111,140 +53,149 @@ let make = (~rawTx, ~onBack, ~account: AccountContext.t) => {
...
-         <div className=Styles.errMsg>
-           <Text value=err color=Colors.red3 align=Text.Center breakAll=true />
+       <div
+         className={Css.merge([
+           CssHelper.flexBox(~direction=`column, ~justify=`center, ()),
+           Styles.resultContainer,
+         ])}>
+         <img src=Images.fail className=Styles.resultIcon />
+         <div className={CssHelper.mb()}>
+           <Text value="Broadcast Transaction Success" size=Text.Lg block=true align=Text.Center />
```
Comment (Pitchanai) [inline scan/src/components/modal/submitTx/PreviewJsonStep.re]: This one too.

**Review by Pitchanai — CHANGES_REQUESTED** [review]

**Review by Pitchanai — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**